builders/ethereum/dev-env/thirdweb.md

```diff
@@ -110,184 +112,339 @@ thirdweb offers SDKs for a range of programming languages, such as React, React
+  value: toWei('1'),
+  // The chain to execute the transaction on. This assumes you already set up
+  // moonbase as a custom chain as shown in the configure chain section.
+  chain: moonbase,
+  // Your thirdweb client
+  client,
+});
+```
 
-    // Your smart contract address
-    const contractAddress = 'INSERT_CONTRACT_ADDRESS';
+### Reading Contract State {: #read-contract-state }
 
-    function App() {
-      const { contract } = useContract(contractAddress);
-      // Read data from your smart contract using the function or variables name
-      const { data, isLoading, error } = useContractRead(contract, 'INSERT_NAME');
-    }
-    ```
+Use the [`readContract` function](https://portal.thirdweb.com/typescript/v5/transactions/read){target=\_blank} to call any read functions on your contract by passing in the Solidity method signature and any parameters.
 
-- Use the [`useContractWrite` hook](https://portal.thirdweb.com/react/react.usecontractwrite){target=\_blank} to call any write functions on your contract by passing in the name of the function you want to use:
-
-    ```javascript
-    import {
-      useContractWrite,
-      useContract,
-      Web3Button,
-    } from '@thirdweb-dev/react';
-
-    // Your smart contract address
-    const contractAddress = 'INSERT_CONTRACT_ADDRESS';
-
-    function App() {
-      const { contract } = useContract(contractAddress);
-      const { mutateAsync, isLoading, error } = useContractWrite(
-        contract,
-        'INSERT_NAME'
-      );
-
-      return (
-        <Web3Button
-          contractAddress={contractAddress}
-          // Calls the 'INSERT_NAME' function on your smart contract
-          // with 'INSERT_ARGUMENT' as the first argument
-          action={() => mutateAsync({ args: ['INSERT_ARGUMENT'] })}
-        >
-          Send Transaction
-        </Web3Button>
-      );
-    }
-    ```
+```typescript title="Reading Contract State"
+import { readContract } from 'thirdweb';
 
-### Connect Wallet {: #connect-wallet }
+const balance = await readContract({
+  contract: contract,
+  method: 'function balanceOf(address) view returns (uint256)',
+  params: ['0x123...'],
+});
+```
 
-There are a couple of ways that you can create a custom [connect wallet](https://portal.thirdweb.com/react/connecting-wallets){target=\_blank} experience. You can use the [`ConnectWallet` component](https://portal.thirdweb.com/react/connecting-wallets#using-the-connect-wallet-button){target=\_blank} or, for a more customizable approach, you can use the [`useConnect` hook](https://portal.thirdweb.com/react/connecting-wallets#using-hooks){target=\_blank}.
+For a function that takes no parameters, such as the number function that returns the current number stored in the [incrementer contract](https://moonbase.moonscan.io/address/0xa72f549a1a12b9b49f30a7f3aeb1f4e96389c5d8){target=\_blank}, you simply need to provide the function name as follows: 
```

Comment:
If we're back to the incrementer example here , I feel like we should just consistently use the incrementer as the example in the other sections